firmware/util/containers/cyclic_buffer.h

```diff
@@ -81,9 +81,7 @@ cyclic_buffer<T, maxSize>::cyclic_buffer(const cyclic_buffer& cb) {
 	currentIndex = cb.currentIndex;
 	count = cb.count;
 	size = cb.size;
-	for (int i = 0; i < size; ++i) {
-		elements[i] = cb.elements[i];
-	}
+	copyArray(elements, cb.elements);
```

Comment:
We may get a small performance penalty here because 'size' could be just 1 and 'maxSize' could be much bigger, so we don't really need to copy the entire array.

Reply:
I'm not sure why we even define the copy constructor.  The default copy constructor simply copies all the fields, which is what we're doing by hand here.

Reply:
(or the copy-assign constructor or destructor, for that matter)